### PR TITLE
FEI-4967.9: Configure build scripts

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
This PR makes the following changes:
- updates `build` script to compile .ts(x) to .js bundles
- adds a `build:types` script which outputs .d.ts files in the dist folders
- updates `publish:ci` to run `build:types`
- fixes all of the TypeScript errors for both `yarn tsc` and `yarn build:types`

Issue: FEI-4967

## Test plan:
- yarn tsc
- yarn build
- yarn build:types
- node ./utils/pre-publish-check-ci.js

### CODE REVIEW

File: config/build/remove-test-types-from-dist.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import * as fglob from "fast-glob";
+
+const rootDir = path.join(__dirname, "..");
```
Comment (jeremywiebe) [inline config/build/remove-test-types-from-dist.ts:6]: isn't the root dir 2 directories up from here?

File: config/build/remove-test-types-from-dist.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    fs.rmdirSync(path.join(rootDir, dir));
+}
+
+const filesInUtils = fglob.sync("utils/**/*.d.ts", {
+    cwd: rootDir,
+});
+
+for (const file of filesInUtils) {
+    fs.unlinkSync(path.join(rootDir, file));
+}
```
Comment (jeremywiebe) [inline config/build/remove-test-types-from-dist.ts:38]: I realize this is just an infra file, but I think it'd be much clearer to read if we extracted a helper function that accepted the globs and an option saying if we're deleting files or directories.  Then the rest of the file could read a bit more declaratively about what we're cleaning up... something like this: 

```suggestion
const deletePaths = (pathGlob, globOptions) => {
    const entries = fglob.sync(pathGlob, {
        cwd: rootDir,
       ...globOptions,
    });

    for (const entry of entries) {
        fs.rm(path.join(rootDir, entry), {recursive: true, force: true});
    }
}

deletePaths("packages/*/dist/**/*.test.d.ts")
deletePaths("packages/*/dist/**/__tests__/*.d.ts")
deletePaths("packages/*/dist/**/__tests__",{onlyDirectories: true})
deletePaths("utils/**/*.d.ts")
```

Comment (kevinbarabash) [reply config/build/remove-test-types-from-dist.ts]: I like this suggestion, sorry I didn't get to use it.

File: config/build/remove-test-types-from-dist.ts (hunk 1 above)
Comment (kevinbarabash) [reply config/build/remove-test-types-from-dist.ts]: Thanks for catching this.  Thankfully we don't need this file any more.

File: config/build/rollup.config.js [hunk 3]
```diff
@@ -199,12 +209,14 @@ const createConfig = (
                 presets: createBabelPresets({platform, format}),
                 plugins: createBabelPlugins({platform, format}),
                 exclude: "node_modules/**",
+                extensions,
             }),
             // This must come after babel() since this plugin doesn't know how
             // to deal with Flow types.
```
Comment (jeremywiebe) [inline config/build/rollup.config.js:215]: ?
```suggestion
            // to deal with TypeScript types.
```

File: packages/perseus/src/widgets/__tests__/radio/focus-ring.test.tsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import {describe, it} from "@jest/globals";
```
Comment (jeremywiebe) [inline packages/perseus/src/widgets/__tests__/radio/focus-ring.test.tsx:1]: I actually don't mind needing to import these Jest symbols.  In my mind, having them implicitly available globally always feels a bit weird.

File: packages/kas/src/__tests__/units.test.ts [hunk 5]
```diff
@@ -44,33 +53,51 @@ expect.extend({
...
             : {
                   pass: false,
                   message: () => `magnitude of ${input} is not ${expected}`,
               };
     },
 });
 
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace jest {
```
Comment (jeremywiebe) [inline packages/kas/src/__tests__/units.test.ts:90]: Having to (re)declare these in each file we use them is a bit saddening. Hopefully we can figure out a way eventually. :)

**Review by jeremywiebe — APPROVED** [review]
This is exciting!

**Pull request merged**